apps/main_app/lib/communicator.ex

```diff
@@ -0,0 +1,34 @@
+defmodule MainApp.Communicator do
+  @moduledoc """
+  Handles communication between M subscribers and N publishers.
```

Comment:
If we are considering `M` subscribers, maybe we could generalize it already in `MainApp` (just like you did with publishers).